From 19736f7dba091c47a9b47ba6903c17a90d758e3f Mon Sep 17 00:00:00 2001
From: Gwenn Bourree <gwenn.bourree@intel.com>
Date: Thu, 12 Mar 2015 15:54:45 +0100
Subject: [PATCH] [Telephony/MCD]: remove code of pmic_io_init

Remove the code of pmic_io_init function because
it is not generic (only applicable for WhiskeyCov)
and causes a regression on SIM detect for CHT_FFD.

Change-Id: Ib3f889f17edf5a8b1b0468016e8d5b6a8ab5faf9
Signed-off-by: Gwenn Bourree <gwenn.bourree@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-6965
---
 drivers/staging/modem_control/mcd_pmic.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/staging/modem_control/mcd_pmic.c b/drivers/staging/modem_control/mcd_pmic.c
index 5020a72..16086d4 100644
--- a/drivers/staging/modem_control/mcd_pmic.c
+++ b/drivers/staging/modem_control/mcd_pmic.c
@@ -43,15 +43,7 @@
 
 int pmic_io_init(void *data)
 {
-	int ret = 0;
-	struct mdm_ctrl_pmic_data *pmic_data = data;
-	u16 addr = pmic_data->chipctrl;
-
-	ret = intel_soc_pmic_writeb(addr, PMIC_MODEMCTRL_REG_RESET);
-	if (ret)
-		pr_err(DRVNAME ": intel_soc_pmic_writeb(ON) failed (err: %d)\n",
-			   ret);
-	return ret;
+	return 0;
 }
 
 int pmic_io_power_on_ctp_mdm(void *data)
-- 
1.9.1

